distributed/asyncio.py

```diff
@@ -191,6 +170,9 @@ def __del__(self):
     replicate = to_asyncio(Client._replicate)
     start_ipython_workers = to_asyncio(Client._start_ipython_workers)
 
+    def __enter__(self):
+        raise RuntimeError("Use AioClient in an 'async with' block, not 'with'")
```

Comment:
Prevents a deadlock if someone does "with AioClient() as client" and it calls the base class's "__enter__".